### PR TITLE
API-1446 log events for products

### PR DESCRIPTION
**Pull request by pyaillet**
Log event for products just after they are sent in the queue

### CODE REVIEW

File: src/Akeneo/Platform/Component/EventQueue/Event.php [hunk 1]
```diff
@@ -50,4 +50,24 @@ public function getUuid(): string
...
+    public function toLog(): array
+    {
+        return [
+            'type' => $this->getName(),
+            'uuid' => $this->uuid,
+            'author' => $this->author->name(),
+            'author_type' => $this->author->type(),
+            'timestamp' => $this->timestamp,
+        ];
+    }
```
Comment (pchasle) [inline src/Akeneo/Platform/Component/EventQueue/Event.php:72]: I'm not a big fan of adding "log" methods to an object like that

Comment (pyaillet) [reply src/Akeneo/Platform/Component/EventQueue/Event.php:72]: Do you think we should handle the conversion logic in each dispatcher ? Or transform this into a dedicated object ?

Comment (pchasle) [reply src/Akeneo/Platform/Component/EventQueue/Event.php:72]: I'm looking at what we can do. Especially as we have a lot logging in the Events API codebase; something we are not used to on the rest of the PIM.

From my point of view, a log depend on a specific context in which it is triggered, as you might need additional informations or a custom structure.

Maybe a simple fix would be to call this method `normalize` instead, that just mean the object will be returned as a standardized associative array.

Comment (willy-ahva) [reply src/Akeneo/Platform/Component/EventQueue/Event.php:72]: I agree with @pchasle . A simple rename in `normalize` is ok for me.

**Review by willy-ahva — APPROVED** [review]
Maybe not in this PR but I see that the dispatchers are not spec. We should do it.

**Comment by pyaillet** [comment]
D'autres remarques !

File: src/Akeneo/Pim/Enrichment/Bundle/EventSubscriber/BusinessEvent/DispatchProductCreatedAndUpdatedEventSubscriber.php [hunk 2]
```diff
@@ -89,6 +89,20 @@ public function dispatchBufferedProductEvents(): void
...
+            foreach ($this->events as $event) {
+                $this->logger->info(
+                    json_encode(
+                        [
+                            'type' => $event->getName(),
+                            'uuid' => $event->getUuid(),
+                            'author' => $event->getAuthor()->name(),
+                            'author_type' => $event->getAuthor()->type(),
+                            'timestamp' => $event->getTimestamp(),
+                        ],
```
Comment (pchasle) [inline src/Akeneo/Pim/Enrichment/Bundle/EventSubscriber/BusinessEvent/DispatchProductCreatedAndUpdatedEventSubscriber.php:101]:
```suggestion
                    json_encode(
                        [
                          'type' => 'event_api.dispatch_event',
                          'event' => [
                                'type' => $event->getName(),
                                'uuid' => $event->getUuid(),
                                'author' => $event->getAuthor()->name(),
                                'author_type' => $event->getAuthor()->type(),
                                'timestamp' => $event->getTimestamp(),
                            ]
                        ],
```
Suggestion on the format

**Review by pchasle — APPROVED** [review]

**Review by tseho — APPROVED** [review]

**Comment by pyaillet** [comment]
@pchasle @willy-ahva we introduced some changes since your reviews, is this still ok for you ?

**Review by pchasle — APPROVED** [review]
I would have used the `event_api` logger directly, instead of creating a new one.

**Comment by pchasle** [comment]
If we follow the ubiquitous language, it should be `pim_event` instead of `business_event`

**Pull request merged**